datafusion/src/sql/planner.rs

```diff
@@ -528,21 +528,43 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
                             name
                         ))),
                     }?,
-                    columns_alias,
+                    alias,
                 )
             }
             TableFactor::Derived {
                 subquery, alias, ..
-            } => (
-                self.query_to_plan_with_alias(
+            } => {
+                // if alias is None, return Err
+                if alias.is_none() {
+                    return Err(DataFusionError::Plan(
+                        "subquery in FROM must have an alias".to_string(),
+                    ));
+                }
+                let logical_plan = self.query_to_plan_with_alias(
                     subquery,
                     alias.as_ref().map(|a| a.name.value.to_string()),
                     ctes,
-                )?,
-                alias.clone().map(|x| x.columns),
-            ),
+                )?;
+                let schema = logical_plan.schema();
+                let mut df_fields_with_alias = Vec::new();
+                for df_field in schema.fields().iter() {
+                    let df_field_with_alias = DFField::from_qualified(
+                        &alias.as_ref().unwrap().name.value,
+                        df_field.field().clone(),
+                    );
+                    df_fields_with_alias.push(df_field_with_alias);
+                }
```

Comment:
it looks like `df_fields_with_alias` is not being used anymore? the schema patching should have been covered by `project_with_alias` right?

Reply:
good sight!